### PR TITLE
Add resource version for object when using CreateAndUpdateObjs

### LINKED ISSUE #47: Modify state.createAndUpdateObjs logic to set resource version for objects
CRD instance needs to modify set resource version for objects on update, which in not currently handled by `state.createAndUpdateObjs`

### PR DESCRIPTION
**Pull request by Mmduh-483**
ResourceVersion is required to update CRD instances

Fixes #47

### CODE REVIEW

File: pkg/state/state_skel.go [hunk 1]
```diff
@@ -112,6 +112,14 @@ func (s *stateSkel) createOrUpdateObjs(
 			return err
 		}
 
+		// Set resource version
```
Comment (adrianchiris) [inline pkg/state/state_skel.go:115]: So, i think this deserves some more info in the comment.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#concurrency-control-and-consistency

Comment (Mmduh-483) [reply pkg/state/state_skel.go:115]: From the doc this value must be passed unmodified back to the server and the  only way for a client to know the expected value of resourceVersion is to have received it from the server

Comment (adrianchiris) [reply pkg/state/state_skel.go:115]: It is used by the kubernetes API server to implement optimistic concurrency for PUT operations.
id like you to add a comment here on why we are setting the resource version so its clear.

**Review by adrianchiris — APPROVED** [review]

**Review by adrianchiris — CHANGES_REQUESTED** [review]
Just add more context to commit message and this can be merged

**Pull request merged**